### PR TITLE
Clean up and add SSD detection...

### PR DESCRIPTION
**Pull request by dak180**
- Add option parsing.
- Add option to specify log and bb files directory.
- Add timestamp to log and bb files.
- Clearer consistent formatting, cleaner logic.
- Add a conveyance test and do not run badblocks on SSDs.
- Check to see if specification of connection type is necessary.
- Check if needed software is installed.
- Try to detect if the device specifier is valid.
- Try to get better disk utilzation from badblocks.

### CODE REVIEW

**Comment by Spearfoot** [comment]
Thanks for the PR, and I apologize for not getting around to looking at it sooner!

I like a lot of your ideas: checking for required tools, adding conveyance testing, checking for SSDs, etc.

I suspect you're using tabs instead of spaces, because when I examine your commit it appears that pretty much every line in the original code has been changed, though it's really just a difference in whitespace. 

I've purposefully avoided `bash`, my idea being to keep things simple and portable. 'Simple' also meaning few bells and whistles like command-line option parsing, etc. 

Can you implement this with the common, garden-variety POSIX shell instead of `bash`? And did you run this through the shellcheck.net validation site?

**Comment by dak180** [comment]
> I suspect you're using tabs instead of spaces, because when I examine your commit it appears that pretty much every line in the original code has been changed, though it's really just a difference in whitespace.

Yes, some of the indenting and line breaks were inconsistent so I standardized them and I like tabs. ☺️ 
 
> I've purposefully avoided `bash`, my idea being to keep things simple and portable. 'Simple' also meaning few bells and whistles like command-line option parsing, etc.
> 
> Can you implement this with the common, garden-variety POSIX shell instead of `bash`? And did you run this through the shellcheck.net validation site?

First, I always use I always use shellcheck.net (it is awesome and everyone should use it ☺️ ); as for the bash vs. sh, while I have not done this yet I want to use local variables in the functions and `local` while supported in _most_ sh variants is not supported in all of them because it is not in POSIX (see [autorevision craziness](https://github.com/Autorevision/autorevision/blob/master/autorevision.sh#L131) as an example), also pure sh does not support arrays. It is easier to just make it bash if you want to do these things than trying to hack it back into sh.

In short, can it be done in sh; yes, but doing it in bash is simpler and less hacky.
